WPMedia/fusion-news-theme-blocks
# ASUB-8195 Sign up with reCaptcha

**Be sure to run `npm test`, `npm run lint`, and write detailed test steps before requesting reviewers**

## Description
This PR includes reCaptcha v2 & v3 for login & Sign Up
Login: https://arcpublishing.atlassian.net/browse/ASUB-8192
Sign Up: https://arcpublishing.atlassian.net/browse/ASUB-8195

The bot-challenge-protection component was moved into [arc-themes-component](ASUB-8193_8192_reCaptchav2_v3) repo, since this component should be shared across different blocks (identity & Subscriptions).

If reCaptcha is enabled the reCaptcha component should be appear

![Screenshot 2024-04-03 at 8 26 33 p m](https://github.com/WPMedia/arc-themes-blocks/assets/54566275/f16320f9-d630-49d6-99b9-9ad479e73d94)

## Test Steps

_Add detailed test steps a reviewer must complete to test this PR_

1. Checkout this branch `git checkout ASUB-8195_SignUp_reCaptcha`
2. For arc-themes component checkout to this branch `git checkout ASUB-8193_8192_reCaptchav2_v3`
2. Run fusion repo with linked blocks `npx fusion start -l @wpmedia/identity-block`
3. Create one page for Login. Includes the "Identity Log In - Arc Block" and set up all the custom fields
4. Create one page for Sign Up. Includes the "Identity Sign Up - Arc Block" and set up all the custom fields
5. For both of them the reCaptcha v2 or v3 should be rendered if it is enabled for the tenant

`org: subssinglesite` `site:the-gazette` `env: sandbox` has recapctha v2 enabled for login & Sign up
         cdn: https://subssinglesite-the-daily-intelligencer-sandbox.api.cdn.arcpublishing.com
         
`org: subssinglesite` `site:the-daily-intelligencer` `env: sandbox` has recapctha v3 enabled for login & Sign up
         cdn: https://subssinglesite-the-daily-intelligencer-sandbox.api.cdn.arcpublishing.com
